### ISSUE TITLE
deltas: updated delta issued after pruned delta

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-uiserver#393]
When a workflow is removed we would expect a single pruned delta with the ID of the workflow.

* pruned(id)

However, we currently get two deltas:

* updated(id), pruned(id)
* updated(id)

The first delta contains an update for the workflow which is erroneous in this case as nothing should have changed in the workflow data, however this update is probably harmless (as we apply updated *before* pruned).

The second delta is more of a problem as it contains updates for a workflow which no longer exists.

This is a major problem for https://github.com/cylc/cylc-ui/pull/1108 which avoids issues like https://github.com/cylc/cylc-ui/issues/903 by treating added & updated deltas as the same, inserting any missing nodes in the process.

<details>
<summary>Steps to reproduce:</summary>

Open GraphiQL to the following subscription:

```graphql
subscription {
  deltas {
    added {
      workflow {
        id
      }
    }
    updated {
      workflow {
        id
      }
    }
    pruned {
      workflow
    }
  }
}
```

Install a workflow (I'm using `cp` rather than `cylc install` to reduce the number of moving parts and ensure the addition is atomic):

```
cp -r ~/cylc-src/my-workflow ~/cylc-run
```

You should get an added delta for the workflow.

Remove the workflow (I'm using `rm` rather than `cylc clean` to reduce the number of moving parts and ensure the removal is atomic):

```
rm -r ~/cylc-run/my-workflow
```

</details>

### Evidence:

![subs](https://user-images.githubusercontent.com/16705946/198253780-5beb2b00-fa95-4cbd-9101-fad3c0794f94.gif)

<img width="969" alt="Screenshot 2022-10-27 at 10 53 06" src="https://user-images.githubusercontent.com/16705946/198253755-f410ad98-1f97-41c0-8fd3-feff7f244993.png">

**Comment by hjoliver** [comment]
Ping @dwsutherland !

**Comment by dwsutherland** [comment]
Unregister was also disconnecting with additional contact update (and hence delta).. #394

**Comment by hjoliver** [comment]
Closed by #394
